pkg/codegen/python/gen.go

```diff
@@ -180,6 +200,35 @@ func (mod *modContext) tokenToEnum(tok string) string {
 	return fmt.Sprintf("'%s%s'", modName, name)
 }
 
+func (mod *modContext) resourceType(r *schema.ResourceType) string {
+	if r.Resource == nil || r.Resource.Package == mod.pkg {
+		return mod.tokenToResource(r.Token)
+	}
+
+	// Is it a provider resource?
+	if strings.HasPrefix(r.Token, "pulumi:providers:") {
+		pkgName := strings.TrimPrefix(r.Token, "pulumi:providers:")
+		return fmt.Sprintf("pulumi_%s.Provider", pkgName)
+	}
+
+	pkg := r.Resource.Package
+	tok := r.Token
+
+	var info PackageInfo
+	contract.AssertNoError(pkg.ImportLanguages(map[string]schema.Language{"python": Importer}))
+	if v, ok := pkg.Language["python"].(PackageInfo); ok {
+		info = v
+	}
+	modName, name := tokenToModule(tok, pkg, info.ModuleNameOverrides), tokenToName(tok)
+	if modName == mod.mod {
+		modName = ""
+	}
+	if modName != "" {
+		modName = strings.ReplaceAll(modName, "/", ".") + "."
+	}
```

Comment:
Would it make sense to pull this into a separate function since it appears in multiple places?